server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -515,12 +530,28 @@ void notifyFailedSnapshotShard(final Snapshot snapshot, final ShardId shardId, f
 
     /** Updates the shard snapshot status by sending a {@link UpdateIndexShardSnapshotStatusRequest} to the master node */
     void sendSnapshotShardUpdate(final Snapshot snapshot, final ShardId shardId, final ShardSnapshotStatus status) {
-        try {
-            UpdateIndexShardSnapshotStatusRequest request = new UpdateIndexShardSnapshotStatusRequest(snapshot, shardId, status);
-            transportService.sendRequest(transportService.getLocalNode(), UPDATE_SNAPSHOT_STATUS_ACTION_NAME, request, INSTANCE_SAME);
-        } catch (Exception e) {
-            logger.warn(() -> new ParameterizedMessage("[{}] [{}] failed to update snapshot state", snapshot, status), e);
-        }
+        remoteFailedRequestDeduplicator.executeOnce(
+            new UpdateIndexShardSnapshotStatusRequest(snapshot, shardId, status),
+            ActionListener.noop(),
+            (req, reqListener) -> {
+                try {
+                    transportService.sendRequest(transportService.getLocalNode(), UPDATE_SNAPSHOT_STATUS_ACTION_NAME, req,
+                        new EmptyTransportResponseHandler(ThreadPool.Names.SAME) {
+                            @Override
+                            public void handleResponse(TransportResponse.Empty response) {
+                                reqListener.onResponse(null);
+                            }
+
+                            @Override
+                            public void handleException(TransportException exp) {
+                                reqListener.onFailure(exp);
+                            }
+                        });
+                } catch (Exception e) {
+                    logger.warn(() -> new ParameterizedMessage("[{}] [{}] failed to update snapshot state", snapshot, status), e);
```

Comment:
Yea, looked into this. This is never hit, the transport logic catches everything (basically just the disconnected exception) and passes it to the listener => moved the logging to the listener since this is just dead code imo.